Trilinos and GMG are broken

@tamiko i think it's related to your recent PRs as it worked just yesterday :smile:
```
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/lac/solver_cg.h: In instantiation of ‘void dealii::SolverCG<VectorType>::solve(const MatrixType&, VectorType&, const VectorType&, const PreconditionerType&) [with MatrixType = dealii::SmartPointer<const dealii::TrilinosWrappers::SparseMatrix, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >; PreconditionerType = dealii::SmartPointer<const dealii::TrilinosWrappers::PreconditionJacobi, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >; VectorType = dealii::TrilinosWrappers::MPI::Vector]’:
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/multigrid/mg_coarse.h:518:3:   required from ‘void dealii::MGCoarseGridIterativeSolver<VectorType, SolverType, MatrixType, PreconditionerType>::operator()(unsigned int, VectorType&, const VectorType&) const [with VectorType = dealii::TrilinosWrappers::MPI::Vector; SolverType = dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>; MatrixType = dealii::TrilinosWrappers::SparseMatrix; PreconditionerType = dealii::TrilinosWrappers::PreconditionJacobi]’
<my_source_file.cc>:105:1:   required from here
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/lac/solver_cg.h:391:11: error: ‘const class dealii::SmartPointer<const dealii::TrilinosWrappers::SparseMatrix, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >’ has no member named ‘vmult’
           A.vmult(g,x);
           ^
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/lac/solver_cg.h:407:11: error: ‘const class dealii::SmartPointer<const dealii::TrilinosWrappers::PreconditionJacobi, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >’ has no member named ‘vmult’
           precondition.vmult(h,g);
           ^
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/lac/solver_cg.h:422:11: error: ‘const class dealii::SmartPointer<const dealii::TrilinosWrappers::SparseMatrix, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >’ has no member named ‘vmult’
           A.vmult(h,d);
           ^
dealii-develop-7zqsrxk7m62idjqctdfa2enpuimjxtxw/include/deal.II/lac/solver_cg.h:440:15: error: ‘const class dealii::SmartPointer<const dealii::TrilinosWrappers::PreconditionJacobi, dealii::MGCoarseGridIterativeSolver<dealii::TrilinosWrappers::MPI::Vector, dealii::SolverCG<dealii::TrilinosWrappers::MPI::Vector>, dealii::TrilinosWrappers::SparseMatrix, dealii::TrilinosWrappers::PreconditionJacobi> >’ has no member named ‘vmult’
               precondition.vmult(h,g);
               ^
```

must be related to https://github.com/dealii/dealii/pull/4606

--- a/include/deal.II/multigrid/mg_coarse.h
+++ b/include/deal.II/multigrid/mg_coarse.h
@@ -515,7 +515,9 @@ MGCoarseGridIterativeSolver<VectorType, SolverType, MatrixType, PreconditionerTy
               const VectorType   &src) const
 {
   Assert(solver!=nullptr, ExcNotInitialized());
-  solver->solve(matrix, dst, src, preconditioner);
+  Assert(matrix!=nullptr, ExcNotInitialized());
+  Assert(preconditioner!=nullptr, ExcNotInitialized());
+  solver->solve(*matrix, dst, src, *preconditioner);
 }
 
 

